From f5512654811e919c87f6f60ab926e145639bfea0 Mon Sep 17 00:00:00 2001
From: Greg Kroah-Hartman <gregkh@suse.de>
Date: Thu, 13 Nov 2008 15:35:52 -0800
Subject: Staging: otus: fix urb callback function type

The #ifdef for version checking is wrong, and we no longer
pass the struct pt_regs to the urb callback functions.  This
patch fixes this error.

Cc: Luis Rodriguez <Luis.Rodriguez@Atheros.com>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>
---
 drivers/staging/otus/wwrap.c |   16 ----------------
 1 file changed, 16 deletions(-)

--- a/drivers/staging/otus/wwrap.c
+++ b/drivers/staging/otus/wwrap.c
@@ -259,11 +259,7 @@ u32_t zfLnxPutUsbRxBuffer(zdev_t *dev, z
     return 0;
 }
 
-#if (LINUX_VERSION_CODE < KERNEL_VERSION(2,5,0))
 void zfLnxUsbDataOut_callback(urb_t *urb)
-#else
-void zfLnxUsbDataOut_callback(urb_t *urb, struct pt_regs *regs)
-#endif
 {
     zdev_t* dev = urb->context;
     //UsbTxQ_t *TxData;
@@ -289,11 +285,7 @@ void zfLnxUsbDataOut_callback(urb_t *urb
     }
 }
 
-#if (LINUX_VERSION_CODE < KERNEL_VERSION(2,5,0))
 void zfLnxUsbDataIn_callback(urb_t *urb)
-#else
-void zfLnxUsbDataIn_callback(urb_t *urb, struct pt_regs *regs)
-#endif
 {
     zdev_t* dev = urb->context;
     struct usbdrv_private *macp = (struct usbdrv_private *)dev->priv;
@@ -526,22 +518,14 @@ void zfLnxUsbDataIn_callback(urb_t *urb,
 #endif
 }
 
-#if (LINUX_VERSION_CODE < KERNEL_VERSION(2,5,0))
 void zfLnxUsbRegOut_callback(urb_t *urb)
-#else
-void zfLnxUsbRegOut_callback(urb_t *urb, struct pt_regs *regs)
-#endif
 {
     //dev_t* dev = urb->context;
 
     //printk(KERN_ERR "zfwUsbRegOut_callback\n");
 }
 
-#if (LINUX_VERSION_CODE < KERNEL_VERSION(2,5,0))
 void zfLnxUsbRegIn_callback(urb_t *urb)
-#else
-void zfLnxUsbRegIn_callback(urb_t *urb, struct pt_regs *regs)
-#endif
 {
     zdev_t* dev = urb->context;
     u32_t rsp[64/4];
